spec/patterns/nagios_spec.rb

```diff
@@ -2,246 +2,463 @@
 require "spec_helper"
 require "logstash/patterns/core"
 
-describe "NAGIOSLOGLINE - CURRENT HOST STATE" do
+describe_pattern "NAGIOSLOGLINE - CURRENT HOST STATE", [ 'legacy', 'ecs-v1' ] do
 
-  let(:value)   { "[1427925600] CURRENT HOST STATE: nagioshost;UP;HARD;1;PING OK - Packet loss = 0%, RTA = 2.24 ms" }
-  let(:grok)    { grok_match(subject, value) }
+  let(:message) { "[1427925600] CURRENT HOST STATE: nagioshost;UP;HARD;1;PING OK - Packet loss = 0%, RTA = 2.24 ms" }
 
   it "a pattern pass the grok expression" do
     expect(grok).to pass
   end
 
   it "matches a simple message" do
-    expect(subject).to match(value)
+    expect(pattern).to match(message)
   end
 
   it "generates the nagios_epoch field" do
-    expect(grok).to include("nagios_epoch" => "1427925600")
+    if ecs_compatibility?
+      expect(grok).to include("timestamp" => "1427925600")
+    else
+      expect(grok).to include("nagios_epoch" => "1427925600")
+    end
   end
 
   it "generates the nagios_message field" do
-    expect(grok).to include("nagios_message" => "PING OK - Packet loss = 0%, RTA = 2.24 ms")
+    if ecs_compatibility?
+      expect(grok).to include("message" => [message, "PING OK - Packet loss = 0%, RTA = 2.24 ms"])
```

Comment:
this artifact feels a little weird to me. I assume this is because we are matching on `[message]` so a capture _appends_ to the existing message?

Reply:
Should we encourage people to _move_ the inbound message to `log.original` before running grok as a general best practice? I assume that would solve this weirdness.

Reply:
Yes, this is a general pattern we're following here when a `message` field is to be replaced due ECS.
This is present on other patterns as well that do "`'message'` replacement".

Not sure at what point to "fix" this, was mostly thinking grok would eventually run with `overwrite => [ "message" ]`.
Maybe, since this is fairly common pattern, we could adopt `overwrite => [ "message" ]` as a default (in ECS mode).

Although I do not like we would have different defaults for a plugin depending on `ecs_compatibility` (but there's already a precedence for that in the ES output plugin).
I was thinking the `.original` copy could happen "earlier" (on event creation) when using an ECS factory ...

Reply:
Yeah copying or moving the original payload in the beginning is a good idea. Note however that ECS unfortunately has 2 such fields, `log.original` and `event.original`. At the next major we're likely to remove `log.original`, so I would recommend using `event.original` instead. Issue for deprecation/removal is here https://github.com/elastic/ecs/issues/841

Reply:
since we have `Event.new("message" => message)` and `"match" => [ "message" => .. ]` hard-coded in spec helper, and also due testing in legacy mode, this seems sufficient for now moving forward. 

have added the concern to our [meta-issue](https://github.com/logstash-plugins/logstash-patterns-core/issues/278)'s post TODO points, as when it gets addressed at a higher level we could get back here and refactor the helpers.